routers/actions_router.js

```diff
@@ -0,0 +1,128 @@
+const express = require("express")
+const router = express.Router()
+
+const actionsDB = require("../data/actions/actions_db")
+const projectsDB = require("../data/projects/projects_db")
+
+router.post("/", (req, res) => {
+  const { action_description, notes, action_completed, project_id } = req.body
+  if (!action_description || !notes || action_completed === null || !project_id)
+    res.status(400).json({
+      errorMessage:
+        "Please provide the action description, notes, completed flag, and project_id."
+    })
+  // Check that project id is valid
+  else {
+    projectsDB.getById(project_id).then(project => {
+      if (project.length > 0) {
+        actionsDB
+          .addAction(req.body)
+          .then(actionId => res.status(201).json(actionId))
+          .catch(err =>
+            res.status(500).json({
+              error:
+                "There was an error while saving the action to the database"
+            })
+          )
+      } else {
+        res.status(400).json({
+          errorMessage: "Please provide a valid id for the project."
+        })
+      }
+    })
+  }
+})
+
+router.get("/", (req, res) => {
+  actionsDB
+    .getAll()
+    .then(actions => {
+      const projects = actions.map(action => action.project_id) // Get all the projects ids
+      const uniqueProjs = projects.filter(
+        (project, index) => projects.indexOf(project) >= index
+      ) // remove all duplicates
+
+      let actionsProj = []
+      uniqueProjs.forEach(proj => {
+        // Build an array of actions that have the same project_id
+        let a = actions.filter(action => action.project_id === proj)
+        a = a.map(action => {
+          return {
+            ...action,
+            completed: action.action_completed ? true : false
+          }
+        })
+        actionsProj.push(a)
```

Comment:
I think this endpoint is more complicated then it needs to be. It's kind of hard to follow what kind of data and structure you were trying to return. I would think returning whatever your `.getAll()` function returns would be fine for this endpoint.